### PR TITLE
:sparkles: Dataset parser

### PR DESCRIPTION
**Pull request by Axelvel**
Added functions to extract the data of an .ecvrp file and create an ECVRPInstance with these parameters.

### CODE REVIEW

**Review by KwassiSenam — APPROVED** [review]
Good job !

File: test/server/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+
+"""
+This module holds tests for the cvrp.ecvrp module.
```
Comment (galyfray) [inline test/server/test_utils.py:4]: Copy paste fail ?

**Review by galyfray — APPROVED** [review]
Very strange situation. But the code is overall good.

File: src/server/utils/utils.py [hunk 2]
```diff
@@ -58,47 +66,58 @@ def parse_dataset(filename: str) -> ECVRPInstance:
...
             elif value == 'STATIONS_COORD_SECTION':
-                for n in range(parameters['STATIONS']):
-                    chargers.add(int(arr[i+n+1]))
+                for index in range(parameters['STATIONS']):
+                    chargers.add(int(arr[i+index+1]))
             else:
                 # Building the parameters dictionnary
                 if not value.isdigit() and value.isupper() and value != 'EOF' \
                         and arr[i+1].replace('.', '', 1).isdigit():
                     val = ast.literal_eval(arr[i+1])
```
[the remark this answers is not in the document]
Comment (Axelvel) [reply src/server/utils/utils.py:84]: Why is that so?

File: test/server/test_utils.py (hunk 1 above)
Comment (Axelvel) [reply test/server/test_utils.py]: Yes indeed 😅

File: src/server/utils/utils.py [hunk 3]
```diff
@@ -58,47 +66,58 @@ def parse_dataset(filename: str) -> ECVRPInstance:
         # Extracting the values
         for i, value in enumerate(arr):
             if value == 'NODE_COORD_SECTION':
-                for n in range((parameters['DIMENSION'])*3):
-                    if n % 3 == 0:
-                        nodes[int(arr[i+n+1])] = (int(arr[i+n+2]), int(arr[i+n+3]))
-                        time_windows[int(arr[i+n+1])] = (0, float('inf'))
+                for index in range((parameters['DIMENSION'])*3):
```
[the remark this answers is not in the document]
Comment (Axelvel) [reply src/server/utils/utils.py:69]: Is that clearer?

File: src/server/utils/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (galyfray) [reply src/server/utils/utils.py:93]: Using the AST to evaluate something might lead to code injection, as far as I know. I didn't dig through the AST module, so I am only asking if you did or if bandit has an opinion on it.

File: src/server/utils/utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (galyfray) [reply src/server/utils/utils.py:76]: yep, better.

**Pull request closed without merging**